MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -1474,54 +1504,141 @@ EnggDiffractionViewQtGUI::splitFittingDirectory(std::string &selectedfPath) {
   return splitBaseName;
 }
 
+void MantidQt::CustomInterfaces::EnggDiffractionViewQtGUI::enableMultiRun(
+    std::string firstRun, std::string lastRun) {
+
+  bool firstDig = isDigit(firstRun);
```

Comment:
All these branches and decisions here are not testable. If you move the logic to the presenter, and add some methods in the view that provide to the presenter what the user has input, then you can do tests with simulated test from the user, and check that the logic is fine and doesn't change unexpectedly in the future.